src/main/java/reposense/RepoSense.java

```diff
@@ -126,16 +126,12 @@ public static void main(String[] args) {
     /**
      * Constructs a list of {@code RepoConfiguration} if {@code cliArguments} is a {@code LocationsCliArguments}.
      */
-    public static List<RepoConfiguration> getRepoConfigurations(LocationsCliArguments cliArguments) {
+    public static List<RepoConfiguration> getRepoConfigurations(LocationsCliArguments cliArguments)
+            throws InvalidLocationException {
         List<RepoConfiguration> configs = new ArrayList<>();
         for (String locationString : cliArguments.getLocations()) {
-            try {
```

Comment:
Thanks for making this contribution!
Removing the try-catch block may not be the most appropriate solution.
I think it'll be helpful if I provide a bit more background so the approach can be improved.
- The `--repo` parameter can accept  multiple repository locations:
  Example: 
`java -jar RepoSense.jar --repo https://github.com/reposense/RepoSense.git https://github.com/se-edu/addressbook-level4.git`
- We would not want to abort RepoSense entirely if only some repository locations are invalid. For example, if the command was `java -jar RepoSense.jar --repo abc.git https://github.com/se-edu/addressbook-level4.git`, then we still want RepoSense to analyse the AddressBook Level 4 repository. RepoSense should continue execution as long as there is a non zero number of valid repository locations. The invalid repository locations can be ignored.

- Currently, we using a logger to emit a warning if any repository location is invalid (line 135 in `RepoSense.java`)

Removing the try-catch block entirely may not be the most appropriate solution as this aborts RepoSense even if there is only one invalid repository location.

Reply:
Thank you for your clarification! So does it mean we need abort RepoSense only if there is one repo location AND the location is invalid?

What is the expected behaviour if there is no argument of location after `--repo`? For example: `java -jar RepoSense.jar --repo`

Reply:
> What is the expected behaviour if there is no argument of location after --repo?

RepoSense is able to detect if 0 locations are provided. A `ParseException` will be thrown and RepoSense will terminate. You can look at the `ArgParser` class to see how the command line arguments are interpreted.

> So does it mean we need abort RepoSense only if there is one repo location AND the location is invalid?

We need to abort RepoSense if all the repo locations are invalid.
Example: If 5 repo locations are provided, and all 5 are invalid, then RepoSense should terminate.
If 5 repo locations are provided, and 4 are invalid, then RepoSense should ignore the 4 invalid locations and proceed to analyze the 1 valid repository.

Reply:
Sure. Thank you so much! I'll try to figure it out.